### PR TITLE
Merge DocExamples.sln

### PR DESCRIPTION
**Pull request by IgorFedchenko**
Continue @Aaronontheweb work on https://github.com/akkadotnet/akka.net/pull/3913

### CODE REVIEW

**Comment by IgorFedchenko** [comment]
@Aaronontheweb why such simple thing might be failing?

```csharp
public class Forwarder : ReceiveActor
{
	private IActorRef target;

	public Forwarder(IActorRef target)
	{
		this.target = target;

		ReceiveAny(target.Forward);
	}
}

[Fact]
public void Test()
{
	//create a test probe
	var probe = CreateTestProbe();
	
	//create a forwarder, injecting the probo's testActor
	var props = Props.Create<Forwarder>(new Forwarder(probe));
	var forwarder = Sys.ActorOf(props, "forwarder");

	//verify correct forwarding
	forwarder.Tell(43, TestActor);
	probe.ExpectMsg(43);
	Assert.Equal(TestActor, probe.LastSender);
}
```

Forwarder even does not receive message, and then `probe.ExpectMsg(43);` is failing. At least on my local machine.

**Comment by Aaronontheweb** [comment]
@IgorFedchenko good question - not sure to be honest. This code has byte-rotted a bit as a result of not being integrated with the main solution for a long time.

**Comment by IgorFedchenko** [comment]
Yea, it was initially expecting `42` instead of `43`, so I doubt that this tests was ever passing :)

**Comment by IgorFedchenko** [comment]
I will let CI run it, just in a case.

**Comment by IgorFedchenko** [comment]
@Aaronontheweb Fixed that test with changing
` var props = Props.Create<Forwarder>(new Forwarder(probe));`
to
`var props = Props.Create(() => new Forwarder(probe));`
I guess this is something about actor creation context

**Comment by Aaronontheweb** [comment]
@IgorFedchenko yeah, sounds like byte rot. This PR will help prevent that in the examples code going forward.

One thing we'll need to update since we're changing the location of the tutorials and tests are all of the DocFx code snippets that reference the current ones: https://dotnet.github.io/docfx/spec/docfx_flavored_markdown.html#code-snippet

You should be able to see which snippets are broken now as a result of running `build.cmd DocFx` and seeing the list of errors / warnings that come back from the DocFx compiler.

CCing @sean-gilliam on this since he's the DocFx master chief around here.

**Comment by IgorFedchenko** [comment]
Ok, I will try to resolve test failures and will check out `build.cmd DocFx` output

**Comment by Aaronontheweb** [comment]
yeah, worry about that only after the thing builds :p

**Comment by Aaronontheweb** [comment]
In fact, why don't you wait to do that until we approve the other changes in this PR first? Because if we need to move things again for whatever reason, it'd suck to have to go through and update all of those.

_Measure twice, cut once_.

**Comment by IgorFedchenko** [comment]
Hmm, if it is possible that you will need to move all of this again... Sure, I will wait for approvement before pushing new updates.

**Comment by Aaronontheweb** [comment]
@IgorFedchenko yeah, once you get the CI system to pass - then we can review it.

**Comment by IgorFedchenko** [comment]
@Aaronontheweb Failures should be resolved now. Now will be wailing for approvement before working on `DocFx` output.

File: src/core/Akka.Docs.Tutorials/Akka.Docs.Tutorials.csproj [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+    <Import Project="..\..\common.props" />
+  <PropertyGroup>
+      <AssemblyTitle>Akka.Docs.Tutorials</AssemblyTitle>
+      <TargetFrameworks>$(NetFrameworkTestVersion);$(NetCoreTestVersion)</TargetFrameworks>
+      <RootNamespace>Tutorials</RootNamespace>
```
Comment (Aaronontheweb) [inline src/core/Akka.Docs.Tutorials/Akka.Docs.Tutorials.csproj]: Need to add the following here:

`<IsPackable>false</IsPackable>`

That will stop us from accidentally publishing an Akka.Docs.Tutorials NuGet package

Comment (IgorFedchenko) [reply src/core/Akka.Docs.Tutorials/Akka.Docs.Tutorials.csproj]: Ok, done.

**Comment by IgorFedchenko** [comment]
`build.cmd DocFx` gives me the following error:

```
Running build failed.
Error:
System.Exception: Restore failed on "restore" "./docs/examples/DocsExamples.sln"
   в Fake.DotNetCli.Restore@146-3.Invoke(String message) в D:\code\fake\src\app\FakeLib\DotNetCLIHelper.fs:строка 146
   в Fake.DotNetCli.Restore(FSharpFunc`2 setRestoreParams) в D:\code\fake\src\app\FakeLib\DotNetCLIHelper.fs:строка 146
   в FSI_0005.Build.clo@551-49.Invoke(Unit _arg11)
   в Fake.TargetHelper.runSingleTarget(TargetTemplate`1 target) в D:\code\fake\src\app\FakeLib\TargetHelper.fs:строка 626
```

Is this `DocExample.sln` hardcoded somewhere? Do you know the possible reason of this error, @Aaronontheweb ?

**Comment by Aaronontheweb** [comment]
Yep, it sure is: https://github.com/akkadotnet/akka.net/blob/7966624e4d66dac5b204416e06b7b4bac1af4ef7/build.fsx#L553-L555

**Comment by Aaronontheweb** [comment]
@IgorFedchenko I'm picking up the thread here to make sure that all of the DocFx files have been updated properly

**Pull request merged**